app/static/tests/e2e/code.etest.ts

```diff
@@ -47,36 +47,40 @@ test.beforeEach(async ({ page }) => {
     await page.goto("/apps/day1");
 });
 
-const expectMonacoDecoration = async (state: EditorStates, line: number, numOfLines: number, page: any) => {
+const expectMonacoDecoration = async (state: EditorStates, line: number, numOfLines: number, page: Page) => {
     for (let i = 0; i < numOfLines; i += 1) {
         const lineElement = await page.locator(`.view-overlays div:nth-child(${i + 1}) >> div`);
-        const glyphElement = await page.locator(`.margin-view-overlays div:nth-child(${i + 1}) >> div`);
         if (i === line - 1) {
-            expect(lineElement).toHaveClass(`cdr editor-${state}-background`);
-            expect(glyphElement.nth(0)).toHaveClass(`cgmr codicon ${editorGlyphs[state]} ${state}-glyph-style ms-1`);
-            expect(glyphElement.nth(1)).toHaveClass("line-numbers lh-odd");
+            await expect(lineElement).toHaveClass(`cdr editor-${state}-background`);
         } else if (i === numOfLines - 1) {
-            expect(lineElement).toHaveClass("current-line");
-            expect(glyphElement.nth(0)).toHaveClass("current-line current-line-margin-both");
-            expect(glyphElement.nth(1)).toHaveClass(/\bactive-line-number\b/);
-            expect(glyphElement.nth(1)).toHaveClass(/\bline-numbers\b/);
-            expect(glyphElement.nth(1)).toHaveClass(/\blh-odd\b/);
+            await expect(lineElement).toHaveClass("current-line");
         } else {
-            expect(lineElement).toHaveCount(0);
-            expect(glyphElement).toHaveClass("line-numbers lh-odd");
+            await expect(lineElement).toHaveCount(0);
         }
     }
+
+    const lineHeight = await page.locator(".margin-view-overlays").evaluate(el => {
+        return window.getComputedStyle(el).getPropertyValue("line-height");
+    });
+    const glyphTop = await page.locator(`.${editorGlyphs[state]}`).evaluate(el => {
+        return window.getComputedStyle(el).getPropertyValue("top");
+    });
+    expect(glyphTop).toBe(`${(line - 1) * parseInt(lineHeight)}px`);
```

Comment:
I feel like this method could use some comments because I'm honestly quite confused about what's being checked for here! In the  loop we're checking that there's only current-line/background styling for `numOfLines`th line and its predecessor? And then the last part is checking that the glyph is on the specified line (by top)..?

Reply:
yep will add some comments but yh the for loop checks for the background colour of the line so the highlighting and this part is checking if the glyph is vertically aligned with the correct line number

Reply:
done